ql/src/java/org/apache/hadoop/hive/ql/parse/GenTezUtils.java

```diff
@@ -76,6 +80,24 @@ public static UnionWork createUnionWork(
     tezWork.add(unionWork);
     return unionWork;
   }
+  
+  private static boolean isRestrictReducerExtrapolation(GenTezProcContext context) {
+
+    Table table = context.parseContext.getContext().getLoadTableOutputMap().values()
+            .stream().findFirst().get().getTable();
+
+    // Iceberg table
+    if (DDLUtils.isIcebergTable(table)) {
```

Comment:
would that work `isIcebergTable(reduceWork.getKeyDesc().getProperties())`?

Reply:
I tried this with a deletion query from Iceberg table on my local using MiniHS2Cluster. 
It didn't work because reduceWork.getKeyDesc() was null.

Reply:
same for `reduceSink.getConf().getKeySerializeInfo()`?

Reply:
Yes, the properties of `reduceSink.getConf().getKeySerializeInfo().properties` don't include info that can help determining table type:

```
serialization.sort.order.null=zzzz, 
columns=reducesinkkey0,reducesinkkey1,reducesinkkey2,reducesinkkey3, serialization.lib=org.apache.hadoop.hive.serde2.binarysortable.BinarySortableSerDe, 
serialization.sort.order=++++, 
columns.types=int,bigint,string,bigint, 
column.name.delimiter=,
```

Reply:
thanks for checking!